Libplanet.Tools/download.js

```diff
@@ -18,64 +24,89 @@ const SUFFIXES = {
   const ver = packageMetadata.version;
   const platform = os.platform();
   const platformSuffixes = SUFFIXES[platform];
+
   if (platformSuffixes == null) {
     throw new Error(`Unsupported platform: ${platform}`);
   }
+
   const arch = os.arch();
   const suffix = platformSuffixes[arch];
+
   if (suffix == null) {
     throw new Error(`Unsupported architecture: ${platform}-${arch}`);
   }
+
   const url = `${URL_BASE}/${ver}/planet-${ver}-${suffix}`;
-  return new Promise((resolve, reject) => {
-    fs.mkdtemp(path.join(os.tmpdir(), "planet-"), (err, dirPath) => {
-      if (err) return reject(err);
-      const finalize = () => {
+
+  return mkdtemp(path.join(os.tmpdir(), "planet-")).then(dirPath => {
+    const finalize = () => {
+      return new Promise((resolve, reject) => {
         const src = path.join(dirPath, BIN_NAME);
         const dst = path.join(binPath, binName);
-        fs.copyFile(src, dst, err => {
-          if (err) return reject(err);
+        
+        return copyFile(src, dst).then(() => {
           if (platform === "win32") return resolve();
           fs.chmod(dst, 0o755, err => err ? reject(err) : resolve());
-        });
-      };
-      let unarchive;
-      if (suffix.toLowerCase().endsWith(".zip")) {
-        unarchive = unzipper.Extract({ path: dirPath });
-        unarchive.on("close", err => {
-          if (err) return reject(err);
-          return finalize();
-        });
-      } else {
-        const subproc = child_process.spawn("tar", ["xvJ"], {
-          cwd: dirPath,
-          stdio: ["pipe", "ignore", "ignore"]
-        });
-        subproc.on("close", code => {
-          if (code !== 0) return reject(code);
-          return finalize();
-        });
-        unarchive = subproc.stdin;
-      }
-      unarchive.on("error", reject);
-      if (log != null) log(`
+        }).catch(reject);
+      });
+    };
+
+    const unarchive = (value) => {
+      return new Promise((resolve, reject) => {
+        let retVal;
+
+        if (suffix.toLowerCase().endsWith(".zip")) {
+          retVal = unzipper.Extract({ path: dirPath });
+          retVal.on("close", err => {
+            if (err) return reject(err);
+            return resolve();
+          });
+        } else {
+          const subproc = child_process.spawn("tar", ["xvJ"], {
+            cwd: dirPath,
+            stdio: ["pipe", "ignore", "ignore"]
+          });
+          subproc.on("close", code => {
+            console.log(code)
+            if (code !== 0) return reject(code);
+            return resolve();
+          });
+          retVal = subproc.stdin;
+        }
+
+        retVal.on("error", reject);
+        value.pipe(retVal);
+      });
+    };
+
+    if (log != null) log(`
 -------------------------------------------------------------------------------
 Downloaing Libplanet CLI Tools ${ver} from GitHub...
-
 NOTE: You can avoid npm entirely by downloading directly from:
-
   ${url}
-
```

Comment:
These empty lines between statements were for readability; could you maintain them?